src/vivarium/framework/results/manager.py

```diff
@@ -90,34 +114,37 @@ def on_post_setup(self, _: Event) -> None:
         missing_stratifications = {}
         unused_stratifications = registered_stratification_names.copy()
 
-        kwargs = {
-            "registered_stratifications": registered_stratifications,
-            "registered_stratification_names": registered_stratification_names,
-            "missing_stratifications": missing_stratifications,
-            "unused_stratifications": unused_stratifications,
-        }
-
         for event_name in self._results_context.observations:
             for (
                 _pop_filter,
-                requested_stratification_names,
+                event_requested_stratification_names,
             ), observations in self._results_context.observations[event_name].items():
+                event_requested_stratification_names = (
+                    set(event_requested_stratification_names)
+                    if event_requested_stratification_names
+                    else set()
+                )
                 for observation in observations:
                     measure = observation.name
-                    kwargs["requested_stratification_names"] = requested_stratification_names
-
-                    df, unused_stratifications = observation.results_initializer(
-                        measure, **kwargs
+                    df = observation.results_initializer(
+                        event_requested_stratification_names, registered_stratifications
                     )
-                    kwargs["unused_stratifications"] = unused_stratifications
                     self._raw_results[measure] = df
-
-        if len(unused_stratifications) > 0:
+                    if observation.stratifications is not None:
+                        unused_stratifications = self._track_stratifications(
```

Comment:
Isn't it simpler to build up the set of stratifications that have actually been used and then compare that to the set of all stratifications that have been registered than to maintain a list of stratifications that haven't been used?

Reply:
I'm indifferent